internal/cmd/cluster/cluster.go

```diff
@@ -25,68 +26,71 @@ var subcommands = []*cli.Command{
 	// Discover(),
 }
 
+var flags = []cli.Flag{
+	&cli.StringSliceFlag{
+		Name:    "addr",
+		Aliases: []string{"a"},
+		Usage:   "static bootstrap `ADDR`",
+		EnvVars: []string{"WW_ADDR"},
+	},
+	&cli.StringFlag{
+		Name:    "discover",
+		Aliases: []string{"d"},
+		Usage:   "bootstrap discovery `ADDR`",
+		Value:   "/ip4/228.8.8.8/udp/8822/multicast/lo0",
+		EnvVars: []string{"WW_DISCOVER"},
+	},
+	&cli.StringFlag{
+		Name:    "ns",
+		Usage:   "cluster namespace",
+		Value:   "ww",
+		EnvVars: []string{"WW_NS"},
+	},
+	&cli.DurationFlag{
+		Name:    "timeout",
+		Usage:   "dial timeout",
+		Value:   time.Second * 15,
+		EnvVars: []string{"WW_CLIENT_TIMEOUT"},
+	},
+}
+
 func Command() *cli.Command {
 	return &cli.Command{
-		Name:    "cluster",
-		Usage:   "cli client for wetware clusters",
-		Aliases: []string{"client"}, // TODO(soon):  deprecate
-		Flags: []cli.Flag{
-			&cli.StringSliceFlag{
-				Name:    "addr",
-				Aliases: []string{"a"},
-				Usage:   "static bootstrap `ADDR`",
-				EnvVars: []string{"WW_ADDR"},
-			},
-			&cli.StringFlag{
-				Name:    "discover",
-				Aliases: []string{"d"},
-				Usage:   "bootstrap discovery `ADDR`",
-				Value:   "/ip4/228.8.8.8/udp/8822/multicast/lo0",
-				EnvVars: []string{"WW_DISCOVER"},
-			},
-			&cli.StringFlag{
-				Name:    "ns",
-				Usage:   "cluster namespace",
-				Value:   "ww",
-				EnvVars: []string{"WW_NS"},
-			},
-			&cli.DurationFlag{
-				Name:    "timeout",
-				Usage:   "dial timeout",
-				Value:   time.Second * 15,
-				EnvVars: []string{"WW_CLIENT_TIMEOUT"},
-			},
-		},
+		Name:        "cluster",
+		Usage:       "cli client for wetware clusters",
+		Aliases:     []string{"client"}, // TODO(soon):  deprecate
+		Flags:       flags,
+		Before:      setup(),
```

Comment:
The setup/teardown functions are already provided for each individual subcommand (see `list`, `publish` and `subscribe`).  If we're going to add them to the `"cluster"`-level subcommand, we should remove them from the lower-level subcommands, otherwise we'll be doing everything twice.

I would recommend just removing them from `"cluster"`, though, because the commented-out `Discover()` command won't be using `Dial`.